ui/app/components/secret-edit.js

```diff
@@ -109,7 +109,7 @@ export default Component.extend(FocusOnInsertMixin, WithNavToNearestAncestor, {
     'mode'
   ),
   canDeleteSecretMetadata: alias('checkMetadataCapabilities.canDelete'),
-  canCreateSecretMetadata: alias('checkMetadataCapabilities.canCreate'),
+  canCreateSecretMetadata: or('checkMetadataCapabilities.canCreate', 'checkMetadataCapabilities.canUpdate'),
```

Comment:
I didn't understand update permissions also always you to set metadata. I double checked.